src/modules/form.js

```diff
@@ -858,75 +858,122 @@ layui.define(['lay', 'layer', 'util'], function(exports){
+          var isCreatable = typeof othis.attr('lay-creatable') === 'string' && isSearch;
+          var isAppendTo = typeof othis.attr('lay-append-to') === 'string';
           var placeholder = optionsFirst
-            ? (optionsFirst.value ? TIPS : (optionsFirst.innerHTML || TIPS))
+            ? (optionsFirst.value ? TIPS : (optionsFirst.innerText || TIPS))
             : TIPS;
 
-          // 替代元素
-          var reElem = $(['<div class="'+ (isSearch ? '' : 'layui-unselect ') + CLASS
-          ,(disabled ? ' layui-select-disabled' : '') + '"></div>'].join(''));
-
-          var triggerElem = $([
-            '<div class="'+ TITLE +'">'
-              ,('<input type="text" placeholder="'+ util.escape($.trim(placeholder)) +'" '
-                +('value="'+ util.escape($.trim(value ? selected.html() : '')) +'"') // 默认值
-                +((!disabled && isSearch) ? '' : ' readonly') // 是否开启搜索
-                +' class="layui-input'
-                +(isSearch ? '' : ' layui-unselect')
-              + (disabled ? (' ' + DISABLED) : '') +'">') // 禁用状态
-              ,'<i class="layui-edge"></i>'
-            ,'</div>'].join(''));
-
-          var contentElem = $(['<dl class="layui-anim layui-anim-upbit'+ (othis.find('optgroup')[0] ? ' layui-select-group' : '') +'">'
-            ,function(options){
+          // 用于替代 select 的外层容器
+          var selectWrapper = (function() {
+            var elem = $('<div class="'+ CLASS +'"></div>');
+            if (!isSearch) {
+              elem.addClass('layui-unselect');
+            }
+            if (disabled) {
+              elem.addClass('layui-select-disabled');
+            }
+            return elem;
+          })();
+
+          var inputElem = (function() {
+            var elem = $('<input type="text" class="layui-input">');
+
+            // 设置占位符和默认值
+            elem.prop('placeholder', $.trim(placeholder));
+            elem.val($.trim(value ? selected.text() : ''));
+
+            // 设置未开启搜索或禁用时的输入框只读状态
+            if (!isSearch || disabled) {
+              elem.prop('readonly', true);
+            }
+
+            // 添加禁用状态时的 className
+            if (disabled) {
+              elem.addClass(DISABLED);
+            }
+
+            return elem;
+          })();
+
+          var titleElem = (function() {
+            var elem = $('<div class="'+ TITLE +'"></div>');
+            elem.append(inputElem);
+            elem.append('<i class="layui-edge"></i>');
+            return elem;
+          })();
+
+          var contentElem = (function() {
+            var elem = $('<dl class="layui-anim layui-anim-upbit"></dl>');
+            if (othis.find('optgroup')[0]) {
+              elem.addClass('layui-select-group');
+            }
+            var content = function() {
               var arr = [];
-              layui.each(options, function(index, item){
+              layui.each(othis.find('*'), function(index, item) {
```

Comment:
可以顺便处理一下这个问题 https://github.com/layui/layui/issues/2336

```suggestion
              layui.each(othis.find('optgroup,option'), function(index, item) {
```